pydicom/tests/test_dataset.py

```diff
@@ -1411,6 +1411,1037 @@ def test_update_with_dataset(self):
         ds2.update(ds)
         assert ds2.PatientName == 'TestC'
 
+    def test_json_roundtrip(self):
+        ds = Dataset()
+        ds.add_new(0x00080005, 'CS', 'ISO_IR 100')
+        ds.add_new(0x00090010, 'LO', 'Creator 1.0')
+        ds.add_new(0x00091001, 'SH', 'Version1')
+        ds.add_new(0x00090011, 'LO', 'Creator 2.0')
+        ds.add_new(0x00091101, 'SH', 'Version2')
+        ds.add_new(0x00091102, 'US', 2)
+
+        jsonmodel = ds.to_json()
+        ds2 = Dataset.from_json(jsonmodel)
+
+        assert ds2.SpecificCharacterSet == ['ISO_IR 100']
+
+    def test_json_private_DS_VM(self):
+        jsonmodel = """{
+    "00080016": {
+        "vr": "UI",
```

Comment:
Perhaps it might make sense to have this as a separate file?  Looks like ~1000 lines.